packages/ado-extension/src/task-config/ado-task-config.ts

```diff
@@ -5,23 +5,32 @@ import * as adoTask from 'azure-pipelines-task-lib/task';
 import { inject, injectable } from 'inversify';
 import { isEmpty } from 'lodash';
 import * as process from 'process';
-import { iocTypes, TaskConfig } from '@accessibility-insights-action/shared';
+import { iocTypes, TaskConfig, TempDirCreator } from '@accessibility-insights-action/shared';
 import normalizePath from 'normalize-path';
 import { resolve } from 'path';
 
 @injectable()
 export class ADOTaskConfig extends TaskConfig {
     constructor(
         @inject(iocTypes.Process) protected readonly processObj: typeof process,
+        @inject(TempDirCreator) private readonly tempDirCreator: TempDirCreator,
         private readonly adoTaskObj = adoTask,
         private readonly resolvePath: typeof resolve = resolve,
     ) {
         super(processObj);
     }
 
+    // memoizing this is important to avoid generating multiple temp directories in the defualt case
+    private memoizedReportOutDir: string | null = null;
     public getReportOutDir(): string {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return this.getAbsolutePath(this.adoTaskObj.getInput('outputDir'))!;
+        if (this.memoizedReportOutDir == null) {
+            const customOutputDir = this.getAbsolutePath(this.adoTaskObj.getInput('outputDir'));
+
+            this.memoizedReportOutDir =
+                customOutputDir != null ? customOutputDir : this.tempDirCreator.createTempDirSync(this.getVariable('Agent.TempDirectory'));
```

Comment:
(small) Would nullish coalescing work here too? It feels a little more readable to me but either is fine.
```suggestion
            this.memoizedReportOutDir =
                customOutputDir ?? this.tempDirCreator.createTempDirSync(this.getVariable('Agent.TempDirectory'));
```

Reply:
Good call; I had avoided it intentionally, thinking that it's not compatible with node < 14, but now that you've suggested it I've remembered that this doesn't matter since we're transpiling it with node10-compatible output settings anyway. (this isn't the first usage in the codebase)